unit_tests/src/test_atomic_apply_acl_operations.cpp

```diff
@@ -636,6 +636,110 @@ TEST_CASE("#7408: Atomic acls respect entity zone")
             }) != e);
 }
 
+TEST_CASE("#7913: Atomic acls accept all acls")
+{
+    using namespace std::string_literals;
+
+    rodsEnv env;
+    _getRodsEnv(env);
+    irods::experimental::client_connection conn;
+    const auto* const user_home = env.rodsHome;
+    const auto* const acl_user_name = "atomic_acl_user";
+
+    ua::user acl_user{acl_user_name};
+
+    // clean up users after end
+    irods::at_scope_exit remove_users{[&] { CHECK_NOTHROW(ua::client::remove_user(conn, acl_user)); }};
+
+    REQUIRE_NOTHROW(ua::client::add_user(conn, acl_user));
+
+    REQUIRE(ua::client::exists(conn, acl_user));
+
+    // create test collection to modify ACLs on
+    const auto test_collection = fs::path{user_home} / "acl_test_collection_issue_7913";
+    REQUIRE_NOTHROW(fs::client::create_collection(conn, test_collection));
+    irods::at_scope_exit remove_test_collection{
+        [&] { REQUIRE(fs::client::remove_all(conn, test_collection, fs::remove_options::no_trash)); }};
+
+    const std::array new_acls_and_perms{std::make_pair("read_metadata", fs::perms::read_metadata),
+                                        std::make_pair("read", fs::perms::read),
+                                        std::make_pair("read_object", fs::perms::read),
```

Comment:
This one is interesting-- since the enums are actually distinct in the `perms` enum, when you `stat` for the permissions, it will actually give you `fs::perms::read` specifically, not `read_object`. If mapped to `read_object`, this test will fail. This also applies to the `modify_object` case!

Reply:
Got it.